apps/mobile/src/components/Community/CommunityPostBottomSheet.tsx

```diff
@@ -0,0 +1,126 @@
+import { useBottomSheetDynamicSnapPoints } from '@gorhom/bottom-sheet';
+import { ForwardedRef, forwardRef, useCallback, useRef } from 'react';
+import { Text, View } from 'react-native';
+import { graphql, useFragment } from 'react-relay';
+import { RefreshIcon } from 'src/icons/RefreshIcon';
+
+import { useToastActions } from '~/contexts/ToastContext';
+import { CommunityPostBottomSheetFragment$key } from '~/generated/CommunityPostBottomSheetFragment.graphql';
+import useSyncTokens from '~/screens/NftSelectorScreen/useSyncTokens';
+
+import { Button } from '../Button';
+import {
+  GalleryBottomSheetModal,
+  GalleryBottomSheetModalType,
+} from '../GalleryBottomSheet/GalleryBottomSheetModal';
+import { useSafeAreaPadding } from '../SafeAreaViewWithPadding';
+import { Typography } from '../Typography';
+
+const SNAP_POINTS = ['CONTENT_HEIGHT'];
+
+type Props = {
+  communityRef: CommunityPostBottomSheetFragment$key;
+  onRefresh: () => void;
+};
+
+function CommunityPostBottomSheet(
+  { communityRef, onRefresh }: Props,
+  ref: ForwardedRef<GalleryBottomSheetModalType>
+) {
+  const community = useFragment(
+    graphql`
+      fragment CommunityPostBottomSheetFragment on Community {
+        name
+        chain
+      }
+    `,
+    communityRef
+  );
+
+  const { bottom } = useSafeAreaPadding();
+  const { isSyncing, syncTokens } = useSyncTokens();
+  const { pushToast } = useToastActions();
+
+  const bottomSheetRef = useRef<GalleryBottomSheetModalType | null>(null);
+
+  const { animatedHandleHeight, animatedSnapPoints, animatedContentHeight, handleContentLayout } =
+    useBottomSheetDynamicSnapPoints(SNAP_POINTS);
+
+  const closeBottomSheet = useCallback(() => {
+    bottomSheetRef.current?.close();
+  }, []);
+
+  const handleSync = useCallback(async () => {
+    if (!community.chain) return;
+
+    await syncTokens(community.chain);
+
+    closeBottomSheet();
+
+    onRefresh();
+    pushToast({
+      message: 'Successfully refreshed your collection',
+    });
+  }, [closeBottomSheet, community.chain, onRefresh, pushToast, syncTokens]);
+
+  return (
+    <GalleryBottomSheetModal
+      ref={(value) => {
```

Comment:
I see we repeat this function a lot, mostly every where that we use `GalleryBottomSheetModal`... do you know what is is for? I wonder if we can reduce the repetition some how

Reply:
I hate this too 😔 

Seems like the latest version bottom sheet provide different types for `animatedSnapPoints`. Upgraded it because of keyboard crash on comment